daemon/create.go

```diff
@@ -139,6 +139,15 @@ func (daemon *Daemon) VolumeCreate(name, driverName string, opts map[string]stri
 		name = stringid.GenerateNonCryptoID()
 	}
 
+	daemon.volumes.mu.Lock()
```

Comment:
Sorry, this is not quite correct.
We can remove all this low-level check here and just check if the volume returned by `daemon.volumes.Create` has the same driver as `driverName`.